projects/bounding_objects/protos/PipeBoundingObject.proto

```diff
@@ -11,79 +12,79 @@ PROTO PipeBoundingObject [
-    elseif subdivision < 8 then
-      io.stderr:write("'subdivision' must be greater than or equal to 8\n")
-      subdivision = 8
-    end
+    // parameter checking
+    let subdivision = fields.subdivision.value;
+    if (subdivision > 200)
+      console.log('High value for \'subdivision\'. This can slow down the simulation');
+    else if (subdivision < 8) {
+      subdivision = 8;
+      console.error('\'subdivision\' must be greater or equal to 8. Value reset to ' + subdivision + '.');
+    }
 
-    local height = fields.height.value
-    if height <= 0 then
-      io.stderr:write("'height' must be greater than 0\n")
-      height = fields.height.defaultValue
-    end
+    let height = fields.height.value;
+    if (height <= 0) {
+      height = fields.height.defaultValue;
+      console.error('\'height\' must be strictly positive. Value reset to ' + height + '.');
+    }
 
-    local radius = fields.radius.value
-    if radius <= 0 then
-      io.stderr:write("'radius' must be greater than 0\n")
-      radius = fields.radius.defaultValue
-    end
+    let radius = fields.radius.value;
+    if (radius <= 0) {
+      radius = fields.radius.defaultValue;
+      console.error('\'radius\' must be strictly positive. Value reset to ' + radius + '.');
+    }
 
-    local thickness = fields.thickness.value
-    if thickness <= 0 then
-      io.stderr:write("'thickness' must be greater than 0\n")
-      thickness = radius / 2
-    elseif thickness >= fields.radius.value then
-      io.stderr:write("'thickness' must be smaller than 'radius'\n")
-      thickness = radius / 2
-    end
+    let thickness = fields.thickness.value;
+    if (thickness <= 0) {
+      thickness = radius * 0.5;
+      console.error('\'thickness\' must be greater than zero. Value reset to ' + thickness + '.');
+    } else if (thickness >= fields.radius.value) {
+      console.error('\'thickness\' must be smaller than \'radius\'.');
+      thickness = radius * 0.5;
+    }
 
-    -- global stuff before entering in the main loop
-    local beta = 2.0 * math.pi / subdivision
-    local alpha = beta / 2.0
-    local innerRadius = radius - thickness
-    local su = radius * math.cos(alpha) - innerRadius
-    if su < 0 then
-      -- fixed edge case:
-      --   There are 2 inner radius, depending if we measure it along the center or along the edge of the boxes.
-      --   If the thickness is below the difference of these two radius, then the algorithm can not achieve.
-      io.stderr:write("Either 'thickness' or 'subdivision' are too small for the box subdivision algorithm.\n")
-      su = math.abs(su)
-    end
-    local sv = height
-    local sw = radius * math.sin(alpha) * 2.0
-    local boxRadius = innerRadius + su / 2.0
-  }%
+    // global stuff before entering in the main loop
+    const beta = 2.0 * Math.PI / subdivision;
+    const alpha = beta * 0.5;
+    const innerRadius = radius - thickness;
+    const su = radius * Math.cos(alpha) - innerRadius;
+    if (su < 0) {
```

Comment:
If su < 0 there is a crash and we don't see your console message:

ERROR: '/home/mael/webots/projects/bounding_objects/protos/PipeBoundingObject.proto': error: Template engine error: failed to execute JavaScript template: .

(no crash in LUA)

Whereas in LUA, there